google/cloud/aiplatform/metadata/schema/base_artifact.py

```diff
@@ -114,6 +115,11 @@ def _init_with_resource_name(
                 Artifact name with the following format, this is globally unique in a metadataStore:
                 projects/123/locations/us-central1/metadataStores/<metadata_store_id>/artifacts/<resource_id>.
         """
+        # Add User Agent Header for metrics tracking if one is not specified
+        # If one is alreayd specified this call was initiated by a sub class.
+        if not base_constants.USER_AGENT_SDK_COMMAND:
+            base_constants.USER_AGENT_SDK_COMMAND = "aiplatform.metadata.schema.base_artifact.BaseArtifactSchema._init_with_resource_name"
```

Comment:
Same comment as above re: storing this prefix as a variable.

Reply:
same comment as above.